Modix.Data/Repositories/MessageRepository.cs

```diff
@@ -166,17 +166,14 @@ public async Task DeleteAsync(ulong messageId)
         public async Task<IReadOnlyList<MessageCountByDate>> GetGuildUserMessageCountByDate(ulong guildId, ulong userId, TimeSpan timespan)
         {
             var earliestDateTime = DateTimeOffset.UtcNow - timespan;
-            var results = await ModixContext.Set<MessageCountByDate>()
-                .FromSqlRaw(
-                    @"select date(""Timestamp""::timestamp AT TIME ZONE 'UTC') as ""Date"", count(""Id"") as ""MessageCount""
+            var results = await ModixContext.Database
+                .SqlQuery<MessageCountByDate>(
+                    $@"select date(""Timestamp""::timestamp AT TIME ZONE 'UTC') as ""Date"", count(""Id"") as ""MessageCount""
                       from ""Messages""
-                      where ""GuildId"" = :GuildId
-                      and ""AuthorId"" = :UserId
-                      and ""Timestamp"" > :StartTimestamp
-                      group by date(""Timestamp""::timestamp AT TIME ZONE 'UTC')",
-                    new NpgsqlParameter(":GuildId", NpgsqlDbType.Bigint) { Value = unchecked((long)guildId) },
-                    new NpgsqlParameter(":UserId", NpgsqlDbType.Bigint) { Value = unchecked((long)userId) },
-                    new NpgsqlParameter(":StartTimestamp", NpgsqlDbType.TimestampTz) { Value = earliestDateTime })
+                      where ""GuildId"" = {unchecked((long)guildId)}
+                      and ""AuthorId"" = {unchecked((long)userId)}
```

Comment:
So with a sufficiently new guild/user this query would soft fail because of the overflow?
Does postgres really not support unsigned longs?

Reply:
It interprets `ulong` as the `decimal` DB type. iirc `long` is interpreted as `bigint`, which has the same range as C# `long`